lasagne/layers/recurrent.py

```diff
@@ -757,3 +762,306 @@ def step_masked(input_dot_W_n, mask_n, cell_previous, hid_previous,
         self.cell_out = cell_out
 
         return hid_out
+
+
+class GRULayer(Layer):
+    """Gated Recurrent Layer (GRU)
+
+    Layer with gated recurrent units implemented as in [1]_, [2]_.
+
+    Parameters
+    ----------
+    incoming : a :class:`lasagne.layers.Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape.
+    num_units : int
+        Number of hidden units.
+    W_in_to_resetgate : function or np.ndarray or theano.shared
+    W_hid_to_resetgate : function or np.ndarray or theano.shared
+    b_resetgate : function or np.ndarray or theano.shared
+    W_in_to_updategate : function or np.ndarray or theano.shared
+    W_hid_to_updategate : function or np.ndarray or theano.shared
+    b_updategate : function or np.ndarray or theano.shared
+    W_in_to_cell : function or np.ndarray or theano.shared
+    W_hid_to_cell : function or np.ndarray or theano.shared
+    b_cell : function or np.ndarray or theano.shared
+    nonlinearity_resetgate : function
```

Comment:
Instead of `function`, it should be `callable or None` (need to fix this in all the layers, so you can ignore that here).